mypy/messages.py

```diff
@@ -992,6 +997,27 @@ def pretty_or(args: List[str]) -> str:
     return ", ".join(quoted[:-1]) + ", or " + quoted[-1]
 
 
+def invariance_notes(notes: List[str], arg_type: str, expected_type: str) -> List[str]:
+    """Explain that the type is invariant and give notes for how to solve the issue."""
+    if expected_type.startswith('List') and arg_type.startswith('List'):
```

Comment:
I think this is a bad idea. Imagine a situation with classes like ``Listener`` or ``DictReader`` (these are actual class names that I have encountered), the note added in this case could be quite confusing.